src/Rule/Nested.php

```diff
@@ -169,4 +168,36 @@ private function fetchOptions(array $rules): array
 
         return $result;
     }
+
+    private static function calculateErrorKey(Rules $aggregatedRule, string $valuePath, string $key): string
+    {
+        if (!self::canConcatenateErrorKey($aggregatedRule, $key)) {
+            return $valuePath;
+        }
+
+        return $valuePath . Result::ERROR_KEY_SEPARATOR . $key;
+    }
+
+    private static function canConcatenateErrorKey(Rules $aggregatedRule, string $key): bool
+    {
+        if (!self::isPositiveInteger($key)) {
+            return true;
+        }
+
+        $aggregatedRuleArr = $aggregatedRule->asArray();
+        if (ArrayHelper::getValueByPath($aggregatedRuleArr, '0.0') !== 'each') {
```

Comment:
That's how it was before. Changed it because it's a bit more compact and readable that way. Not sure performance gain will be big enough here (without benchmarks). As always, performance vs readability ("design first, optimize later" principle). But OK, I'll change it back.

Reply:
:heavy_check_mark: